.github/workflows/matrix.yml

```diff
@@ -0,0 +1,62 @@
+name: Test Matrix
+on:
+  push:
+    branches:
+      - develop
+      - master
+  pull_request:
+    branches:
+      - develop
+      - master
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version: [16, 18, 20]
+        operating-system:
+          - ubuntu-latest
+          - windows-latest
+          # - macos-latest -> Rollup is not supported on ARM64 - https://github.com/npm/cli/issues/4828
+    runs-on: ${{ matrix.operating-system }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install and Cache Dependencies
+        uses: ./.github/actions/cached-deps
+      - name: Build
+        run: npm run build
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version: [16, 18, 20]
+        operating-system: [ubuntu-latest, windows-latest, macos-latest]
+    runs-on: ${{ matrix.operating-system }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install and Cache Dependencies
+        uses: ./.github/actions/cached-deps
+      - name: Test
+        run: npm run test
+
+  check:
```

Comment:
I figured out how to fail the build if the testing matrix fails at any step. I'm also open to changing this too.